graphqlapi/dynamic_generation_meta_fetch.go

```diff
@@ -0,0 +1,650 @@
+
+	topOccurencesFields, err := genMetaClassStringPropertyTopOccurrencesFields(class, property)
+	if err != nil {
+		return nil, err
+	}
+	metaFetchMetaPointingFields := graphql.Fields{
+		"type": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "MetaPointingTo"),
+			Description: "datatype of the property",
+			Type:        graphql.Int,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"counter": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "MetaPointingFrom"),
+			Description: "total amount of found instances",
+			Type:        graphql.Int,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"topOccurences": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, "MetaPointingFrom"),
+			Description: "most frequent property values",
+			Type:        topOccurencesFields,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+	}
+	metaFetchStringProperty := graphql.ObjectConfig{
+		Name:        mergeStrings("Meta", class.Class, property.Name, "Obj"),
+		Fields:      metaFetchMetaPointingFields,
+		Description: "Property meta information",
+	}
+	return graphql.NewObject(metaFetchStringProperty), nil
+}
+
+func genMetaClassStringPropertyTopOccurrencesFields(class *models.SemanticSchemaClass, property *models.SemanticSchemaClassProperty) (*graphql.Object, error) {
+
+	metaFetchMetaPointingFields := graphql.Fields{
+		"value": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, property.Name, "TopOccurencesValue"),
+			Description: "property value of the most frequent properties",
+			Type:        graphql.String,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+		"occurs": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, property.Name, "TopOccurencesOccurs"),
+			Description: "number of occurrance",
+			Type:        graphql.String,
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				return nil, fmt.Errorf("Not supported")
+			},
+		},
+	}
+	metaFetchMetaPointing := graphql.ObjectConfig{
+		Name:        mergeStrings("Meta", class.Class, property.Name, "TopOccurencesObj"),
+		Fields:      metaFetchMetaPointingFields,
+		Description: "most frequent property values",
+	}
+	return graphql.NewObject(metaFetchMetaPointing), nil
+}
+
+func genMetaClassIntPropertyFields(class *models.SemanticSchemaClass, property *models.SemanticSchemaClassProperty) (*graphql.Object, error) {
+
+	metaFetchMetaIntFields := graphql.Fields{
+		"sum": &graphql.Field{
+			Name:        mergeStrings("Meta", class.Class, property.Name, "Sum"),
+			Description: "sum of values of found instances",
+			Type:        graphql.Float,
```

Comment:
Should this not be an int? Given that we're interested in summing up ints, an int is sufficient.

Reply:
Same applies to other fields, except `average`, that probably should remain a float.

Reply:
Good catch!